crates/sozo/src/commands/options/mod.rs

```diff
@@ -1,10 +1,31 @@
-use scarb::core::Workspace;
+use scarb::core::{ManifestMetadata, Workspace};
 use toml::Value;
 
 pub mod account;
 pub mod starknet;
 pub mod world;
 
-pub(super) fn dojo_metadata_from_workspace(ws: &Workspace<'_>) -> Option<Value> {
-    ws.current_package().ok()?.manifest.metadata.tool_metadata.as_ref()?.get("dojo").cloned()
+pub(crate) fn dojo_metadata_from_workspace(ws: &Workspace<'_>) -> Option<DojoMetadata> {
+    Some(ws.current_package().ok()?.manifest.metadata.dojo_metadata())
+}
+
+pub(crate) struct DojoMetadata {
+    env: Option<Value>,
```

Comment:
Lets have `env` to have a defined type as well instead of just `Value`. We already know what fields may be in the Dojo metadata which you can find under `sozo::commands::options::*`, all the options args.

```rust
struct Environment {
   rpc_url: Option<String>,
   ...
}
```